File: app/src/main/java/com/example/data/dao/HabitStreakDao.java
package com.example.data.dao;

import androidx.room.*;
import com.example.data.model.HabitStreak;
import kotlinx.coroutines.flow.Flow;
import java.util.List;

@Dao
public interface HabitStreakDao {
    @Query("SELECT * FROM habit_streaks")
    Flow<List<HabitStreak>> getAllStreaks();

    @Query("SELECT * FROM habit_streaks WHERE taskType = :taskType")
    HabitStreak getStreakByTaskType(String taskType);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertOrUpdateStreak(HabitStreak streak);
}

File: app/src/main/java/com/example/data/dao/HabitStreakDao.kt
package com.example.data.dao

import androidx.room.*
import com.example.data.model.HabitStreak
import kotlinx.coroutines.flow.Flow

@Dao
interface HabitStreakDao {
    @Query("SELECT * FROM habit_streaks")
    fun getAllStreaks(): Flow<List<HabitStreak>>

    @Query("SELECT * FROM habit_streaks WHERE taskType = :taskType")
    suspend fun getStreakByTaskType(taskType: String): HabitStreak?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertOrUpdateStreak(streak: HabitStreak)
}
